samples/sid_end_device/src/cli/sbdt_shell.c

```diff
@@ -0,0 +1,790 @@
+
+struct sbdt_context {
+	struct sid_bulk_data_transfer_event_callbacks ft_callbacks;
+	uint32_t finalize_response_delay_s;
+	uint32_t release_buffer_delay_ms;
+	uint8_t finalize_response_action;
+	uint8_t transfer_request_action;
+	enum sid_bulk_data_transfer_reject_reason transfer_request_reject_reason;
+	bool started_transfer;
+};
+
+static struct sbdt_context sbdt_context = {
+        .ft_callbacks = {
+                .context = NULL,
+                .on_transfer_request = on_sbdt_transfer_request,
+                .on_data_received = on_sbdt_data_received,
+                .on_finalize_request = on_sbdt_finalize_request,
+                .on_cancel_request = on_sbdt_cancel_request,
+                .on_error = on_sbdt_error,
+                .on_release_scratch_buffer = on_release_scratch_buffer,
+                },
+        .transfer_request_action = 0,
+        .transfer_request_reject_reason = 0x1,
+        .finalize_response_action = 0,
+        .finalize_response_delay_s = 1,
+        .release_buffer_delay_ms = 30,
+	.started_transfer = false,
+};
+
+/* --- event declaration - to be moved to separate header --- */
+#include <sidewalk.h>
+
+void sbdt_event_init(sidewalk_ctx_t *sid, void *ctx);
+void sbdt_event_deinit(sidewalk_ctx_t *sid, void *ctx);
+void sbdt_event_cancel(sidewalk_ctx_t *sid, void *ctx);
+
+void sbdt_event_print_stats(sidewalk_ctx_t *sid, void *ctx);
+void sbdt_event_print_params(sidewalk_ctx_t *sid, void *ctx);
+void sbdt_event_finalize_request_response(sidewalk_ctx_t *sid, void *ctx);
+void sbdt_event_release_buffer(sidewalk_ctx_t *sid, void *ctx);
+
+struct sbdt_cancel_ctx {
+	int file_id;
+	enum sid_bulk_data_transfer_reject_reason reason;
+};
+
+struct sbdt_finalize_resp_ctx {
+	int file_id;
+	uint8_t finalize_response_action;
+	struct k_timer delay;
+};
+struct sbdt_buffer_release_ctx {
+	int file_id;
+	struct sid_bulk_data_transfer_buffer transfer_buffer;
+	struct k_timer delay;
+};
+
+/* --- shell handlers --- */
+
+static int cmd_sbdt_init(const struct shell *shell, int32_t argc, const char **argv)
+{
+	sidewalk_event_send(sbdt_event_init, &sbdt_context.ft_callbacks, NULL);
+	return 0;
+}
+
+static int cmd_sbdt_deinit(const struct shell *shell, int32_t argc, const char **argv)
+{
+	sidewalk_event_send(sbdt_event_deinit, NULL, NULL);
+	return 0;
+}
+
+static bool validate_reason(int reason_i)
+{
+	switch (reason_i) {
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_NONE:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_GENERIC:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_FILE_TOO_BIG:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_NO_SPACE:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_LOW_BATTERY:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_FILE_VERIFICATION_FAILED:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_FILE_ALREADY_EXISTS:
+	case SID_BULK_DATA_TRANSFER_REJECT_REASON_INVALID_FRAGMENT_SIZE:
+		return true;
+		// intentonally left without default to warn about new values
```

Comment:
Please remove this comment before moving from draft

Reply:
removed